capabilities/index/index.go

```diff
@@ -0,0 +1,89 @@
+package index
+
+import (
+	"fmt"
+
+	"github.com/ipfs/go-cid"
+	"github.com/storacha/go-libstoracha/capabilities/types"
+	"github.com/storacha/go-ucanto/core/result/failure"
+	"github.com/storacha/go-ucanto/core/schema"
+	"github.com/storacha/go-ucanto/did"
+	"github.com/storacha/go-ucanto/ucan"
+	"github.com/storacha/go-ucanto/validator"
+)
+
+// SpaceDID represents a DID of a space
+type SpaceDID did.DID
+
+// IndexArgs represents the arguments for the space/index/add capability
+type IndexArgs struct {
+	// Link is the Content Archive (CAR) containing the `Index`.
+	Index cid.Cid `json:"index"`
```

Comment:
we usually define these kind of fields as `ucan.Link` rather than plain `cid.Cid`s.

Also, is the json struct tag useful for anything? We are not using a JSON encoder/decoder with these structs.